checker/checker.go

```diff
@@ -322,65 +322,64 @@ func (c *checker) checkBlockStmt(scope *parser.Scope, typ parser.ObjType, block
 			panic("implementation error")
 		}
 
-		lookupByType := builtin.Lookup.ByType[typ]
-		if !foundSource {
-			// If the function is not a builtin, retrieve it from the scope and then
-			// type check it.
-			_, ok := lookupByType.Func[name]
-			if !ok {
-				obj := scope.Lookup(name)
-				if obj == nil {
-					return ErrFirstSource{call}
-				}
+		// If the function is not a builtin, retrieve it from the scope and then
+		// type check it.
+		funcLookup, ok := builtin.Lookup.ByType[typ].Func[name]
```

Comment:
This appears safe since we are not dealing with any pointer types, I changed the code to this and it ran without exceptions:
```
funcLookup, ok := builtin.Lookup.ByType["foobar"].Func[name]
if ok {
```